code/game/verbs/ooc.dm

```diff
@@ -57,24 +57,38 @@
 		display_colour = CONFIG_GET(string/ooc_color_default)
 
 	msg = process_chat_markup(msg, list("*"))
-
+	var/ooc_prefix = handle_ooc_prefix()
 	for(var/client/C in GLOB.clients)
 		if(C.prefs.toggles_chat & CHAT_OOC)
 			var/display_name = src.key
 			if(prefs.unlock_content)
 				if(prefs.toggle_prefs & TOGGLE_MEMBER_PUBLIC)
 					var/byond = icon('icons/effects/effects.dmi', "byondlogo")
-					display_name = "[icon2html(byond, GLOB.clients)][display_name]"
+					ooc_prefix = "[icon2html(byond, GLOB.clients)][ooc_prefix]"
 			if(CONFIG_GET(flag/ooc_country_flags))
 				if(prefs.toggle_prefs & TOGGLE_OOC_FLAG)
-					display_name = "[country2chaticon(src.country, GLOB.clients)][display_name]"
-			to_chat(C, "<font color='[display_colour]'><span class='ooc linkify'>[src.donator ? "\[D\] " : ""]<span class='prefix'>OOC: [display_name]</span>: <span class='message'>[msg]</span></span></font>")
+					ooc_prefix = "[country2chaticon(src.country, GLOB.clients)][ooc_prefix]"
```

Comment:
All of this should go into `handle_ooc_prefix()`. Otherwise you are adding membership icon and flag once _per everybody listening_.
![image](https://github.com/cmss13-devs/cmss13/assets/4447185/363a00a1-4129-43fb-882a-954358fd624d)